### PR TITLE
[FLINK-3439] Remove final Long.MAX_VALUE Watermark in StreamSource

### PR DESCRIPTION
**Pull request by aljoscha**

### CODE REVIEW

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/timestamp/TimestampITCase.java [hunk 1]
```diff
@@ -157,13 +157,6 @@ public void testWatermarkPropagation() throws Exception {
 					Assert.fail("Wrong watermark.");
 				}
 			}
-			if (!CustomOperator.finalWatermarks[i].get(CustomOperator.finalWatermarks[i].size() - 1).equals(new Watermark(Long.MAX_VALUE))) {
```
Comment (StephanEwen) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/timestamp/TimestampITCase.java]: Let's add a test here that validates that this `Long.MAX_VALUE` watermark is not present. Something like `assertEquals(NUM_WATERMARKS, CustomOperator.finalWatermarks[i].size())`

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/timestamp/TimestampITCase.java [hunk 2]
```diff
@@ -346,9 +339,6 @@ public long checkAndGetNextWatermark(Integer element, long extractedTimestamp) {
 				Assert.fail("Wrong watermark.");
 			}
 		}
-		if (!CustomOperator.finalWatermarks[0].get(NUM_ELEMENTS).equals(new Watermark(Long.MAX_VALUE))) {
```
Comment (StephanEwen) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/timestamp/TimestampITCase.java]: Check like above

**Comment by StephanEwen** [comment]
Good and important fix. Can you add checks as described in the comments?

Otherwise, +1 to merge

**Comment by aljoscha** [comment]
I added the checks, will merge later once travis runs through.

**Pull request closed without merging**